src/api/eventTypes.js

```diff
@@ -173,3 +175,236 @@ export type RestartEvent = $ReadOnly<{|
   zulip_version?: string,
   zulip_feature_level?: number,
 |}>;
+
+/* eslint-disable flowtype/generic-spacing */
+
+/**
+ * Data updates we might get in `update` or `update_dict` realm events.
+ */
+// If there's a property that exists in InitialDataRealm, but we don't use
+// it and the event doc doesn't mention it, write the InitialDataRealm
+// property in a code comment in the place where it would go here.
+type RealmDataForUpdate = $ReadOnly<{
+  //
+  // Keep alphabetical by the InitialDataRealm property. So by
+  // realm_allow_edit_history, not by allow_edit_history, etc.
+  //
+
+  development_environment: $PropertyType<InitialDataRealm, 'development_environment'>,
+
+  // TODO(server-5.0): See note in InitialDataRealm.
+  // event_queue_longpoll_timeout_seconds
+
+  jitsi_server_url: $PropertyType<InitialDataRealm, 'jitsi_server_url'>,
+  max_avatar_file_size_mib: $PropertyType<InitialDataRealm, 'max_avatar_file_size_mib'>,
+  max_file_upload_size_mib: $PropertyType<InitialDataRealm, 'max_file_upload_size_mib'>,
+
+  // TODO: Check if the doc really means this, instead of max_icon_file_size
+  // and max_icon_file_size_mib?
+  icon_file_size: number,
+
+  // TODO(server-5.0): See notes in InitialDataRealm.
+  // max_logo_file_size
+  // max_logo_file_size_mib
+
+  password_min_length: $PropertyType<InitialDataRealm, 'password_min_length'>,
+  password_min_guesses: $PropertyType<InitialDataRealm, 'password_min_guesses'>,
+
+  // TODO(server-5.0): See notes in InitialDataRealm.
+  // realm_add_custom_emoji_policy
+  // realm_add_emoji_by_admins_only
+  // realm_allow_community_topic_editing
+
+  allow_edit_history: $PropertyType<InitialDataRealm, 'realm_allow_edit_history'>,
```

Comment:
Hmm, using `$PropertyType` is a good idea here!

Reply:
Yep! We'll get a nice shortening of all these in a future Flow version: https://flow.org/en/docs/types/utilities/#toc-propertytype